node/src/fork.rs

```diff
@@ -0,0 +1,158 @@
+//! Support for forking off another client
```

Comment:
Why can we not re-use the Forked `cache` from `foundry-evm` here?

Reply:
it is used, but we need to combine the `SharedBackend`and `MemDb` in order to have the full range revm's Database traits